packages/emotion/src/withStyle.tsx

```diff
@@ -31,6 +31,20 @@ import { useTextDirectionContext } from '@instructure/ui-i18n'
 import { bidirectionalPolyfill } from './styleUtils/bidirectionalPolyfill'
 import { getComponentThemeOverride } from './getComponentThemeOverride'
 import { useTheme } from './useTheme'
+
+export interface MakeStyleConfig {
+  // The previous prop object that will be compared to the current props
+  prevProps: Record<string, unknown>
+  // An array of property names used in styles.js. The style object will
+  // recalculate if any of these properties change In cannot be used to compare
+  // nested properties.
+  styledProps: string[]
+}
+export type MakeStyles = (
+  generateStyleArgs?: Record<string, unknown>,
+  updateConfig?: MakeStyleConfig
+) => void
```

Comment:
I'm not sure about this function signature, it could lead to ugly call sites like `makeStyles(undefined, config)`. 
If we keep backwards compatibly in mind and want to preserve the 1 parameter call signature, then we could do something like this:
```js
export type MakeStyleConfig = {
  updateConfig?: {
    prevProps: Record<string, unknown>
    styledProps: string[]
  }
  [K: string]: unknown
}
export type MakeStyles = (config: MakeStyleConfig) => void
```